distributed/scheduler.py

```diff
@@ -711,6 +727,9 @@ def remove_worker(self, stream=None, address=None):
             if address in self.saturated:
                 self.saturated.remove(address)
 
+            for worker_sets in self.environment_workers.values():
+                worker_sets.discard(address)
```

Comment:
Can you add a test inside of `Scheduler.validate` to check that all workers in `self.environment_workers` are also in`self.worker_info`.  Don't bother with `scheduler.py:validate()` (which is the prominent function in Scheduler.validate)